scripts/solve_network.py

```diff
@@ -814,6 +814,36 @@ def add_pipe_retrofit_constraint(n):
     n.model.add_constraints(lhs == rhs, name="Link-pipe_retrofit")
 
 
+def add_ocgt_retrofit_constraint(n):
+    """
+    Add constraint for retrofitting existing OCGT to H2 OCGT.
+    """
+    # existing OCGT plants
+    gas_i = n.links.query(
+        "carrier == 'OCGT' and p_nom_extendable and p_nom > 1e-3"
```

Comment:
maybe I am wrong, but shouldn't links with small capacities (<10 MW) be filtered out already in add_brownfield?

Reply:
Very strange error when I remove the condition:
I seems like the variable `ocgt_i` is not empty 
``` 
('CZ1 0 OCGT-2020', 'DE1 0 OCGT-2020', 'DE1 1 OCGT-2020',
'DE1 2 OCGT-2020', 'DE1 3 OCGT-2020', 'DE1 4 OCGT-2020',
'DE1 5 OCGT-2020', 'DE1 6 OCGT-2020', 'DE1 7 OCGT-2020',
'DK1 0 OCGT-2020', 'DK2 0 OCGT-2020', 'FR1 0 OCGT-2020',
'GB0 0 OCGT-2020', 'GB3 0 OCGT-2020', 'LU1 0 OCGT-2020',
'NL1 0 OCGT-2020', 'NO2 0 OCGT-2020', 'PL1 0 OCGT-2020',
'SE2 0 OCGT-2020')
 ```
with capacities ~ 1e-11. I'll discuss what's happening.
 Error:
```
File "/home/toni-seibold/Documents/02_repos/pypsa-ariadne/workflow/submodules/pypsa-eur/scripts/add_existing_baseyear.py", line 654, in add_ocgt_retro
    df.loc[:, "capital_cost"] *= retro_factor_ocgt
  File "/home/toni-seibold/micromamba/envs/pypsa-ariadne/lib/python3.11/site-packages/pandas/core/generic.py", line 12729, in __imul__
    return self._inplace_method(other, type(self).__mul__)  # type: ignore[operator]
```
I can't reproduce the error when being in an interactive window

Reply:
@fneum and I found a bug that led to this strange behavior 
https://github.com/PyPSA/pypsa-eur/commit/450f75a989dd9b84e27a7690a96107bdca4e8035